### PR TITLE
Version 1.0.1

### PR DESCRIPTION
**Pull request by mahmoudElkarargyBS**
## :bulb: Brief
Version 1.0.1 which mainly fix bugs
 
## :clipboard: Task description from last release v1.0.0
* [x] Implemented a new changelog.
* [x] Introduced a benchmarking script.
* [x] Resolved issues with MPI installation.
* [x] Fixed the printing of configuration summaries with MPI.
* [x] Automated the process of adding a new kernel.
* [x] Improved packaging of software with CPack.
* [x] Addressed installation issues.
* [x] Fixed bivariate and trivariate kernel functionality.
* [x] Corrected time-space kernel issues.
* [x] Updated the installation process for dependencies.
* [x] Modified the calculation of P for all kernels.
* [x] Adjusted CMake variables.
* [x] Revised the process of finding BlasPP and Catch2 libraries.
* [x] Updated doxygen documentation.
* [x] Eliminated non-stationary kernel support.
* [x] Removed Find OpenMP, LapackPP, and CuSolver.

## :checkered_flag: Merge Request Code Quality Checklist

- [x] **Code Review**: Request a thorough code review from at least one team member.
- [x] **Coding Standards**: Ensure the code follows the established coding standards and style guide (e.g., Google Style Conventions).
- [x] **Naming Conventions**: Verify that variable, function, and class names are clear, descriptive, and follow the naming conventions.
- [x] **Comments**: Check for clear and concise comments explaining the code's purpose, logic, and any complex algorithms or business rules.
- [x] **Function/Method Documentation**: Ensure that important functions and methods are properly documented using Doxygen or a similar documentation tool.
- [x] **Code Formatting**: Confirm consistent code formatting, including proper indentation, spacing, and line breaks.
- [x] **Error Handling**: Check for appropriate error handling and meaningful error messages.
- [x] **Testing**: Verify that the code has been adequately tested, including unit tests and integration tests, to cover different scenarios and edge cases.
- [x] **Test Coverage**: Validate that the code has sufficient test coverage, aiming for a high percentage of code coverage.
- [x] **Performance**: Consider the code's performance implications and identify any potential bottlenecks or optimizations.
- [x] **Code Duplication**: Look for duplicated code and refactor it into reusable functions or classes.
- [x] **Dependencies**: Ensure that new dependencies are necessary and properly documented, and that existing dependencies are up to date.
- [x] **Memory Management**: Check for proper memory allocation and deallocation, avoiding memory leaks and excessive memory usage.
- [x] **Thread Safety**: Consider thread safety concerns and ensure proper synchronization mechanisms are in place, if applicable.
- [x] **Documentation**: Update any relevant documentation, including README files, user guides, and API documentation.
- [x] **Performance**: Consider the code's performance implications and identify any potential bottlenecks or optimizations.
- [x] **Integration**: Ensure the code integrates smoothly with existing components and systems.
- [x] **Build**: Verify that the code builds successfully, without errors or warnings, on the target platforms and configurations.
- [x] **Code Review Feedback**: Address any feedback received during the code review process and make necessary changes.
- [x] **Testing**: Perform additional testing after incorporating code review feedback.

## :pencil: C++ Best Practices

Please consider the following best practices when working with C++:

- **Use Modern C++ Features**: Utilize modern C++ features and idioms, such as smart pointers, lambda expressions, range-based loops, and standard library algorithms.
- **Follow Coding Standards**: Adhere to a consistent coding style and follow established coding standards, such as the Google C++ Style Guide or the C++ Core Guidelines.
- **Avoid Using C-Style Constructs**: Minimize the use of C-style constructs (e.g., raw pointers, arrays) and prefer safer and more expressive C++ alternatives.
- **RAII (Resource Acquisition Is Initialization)**: Utilize the RAII principle to manage resources automatically through constructors and destructors, ensuring proper resource cleanup.
- **Avoid Global Variables**: Minimize the use of global variables as they can lead to tight coupling and make code harder to test and maintain. Prefer encapsulation and proper scoping.
- **Avoid Magic Numbers**: Replace magic numbers with named constants or enums to improve code readability and maintainability.
- **Use Meaningful Names**: Choose descriptive and meaningful names for variables, functions, classes, and other identifiers to enhance code understandability.
- **Avoid Code Duplication**: Refactor duplicated code into reusable functions or classes to promote code reuse and reduce maintenance efforts.
- **Error Handling**: Use appropriate error handling mechanisms, such as exceptions or error codes, to handle and propagate errors consistently throughout the codebase.
- **Unit Testing**: Write comprehensive unit tests to validate the correctness of individual components and ensure proper code coverage.
- **Avoid Undefined Behavior**: Be aware of undefined behavior in C++ and avoid writing code that produces undefined or unspecified results.
- **Memory Management**: Use appropriate memory management techniques, such as smart pointers or containers, to avoid memory leaks and dangling pointers.
- **Const-Correctness**: Utilize `const` and `constexpr` to enforce immutability and prevent unintended modifications to variables and parameters.
- **Keep Code Simple**: Strive for simplicity in code design and implementation, favoring clear and maintainable solutions over unnecessarily complex ones.

These best practices aim to enhance code readability, maintainability, and reliability while following industry-standard guidelines and principles.

## :arrow_down: Installation
Please follow up on the README.md and USER_MANUAL.md for the installation, It will be a great test to check the readability of the README and to rate how easy you can download our software

## :copyright: Copyrights
 Developer: @mahmoud.elkarargy

 Reviewers: @omar.marzouk and @merna.moawad.
 
`
 Copyright (c) 2017-2023 King Abdullah University of Science and Technology,
 `

 `
 Copyright (C) 2021-2023 by Brightskies Inc.
 `

 `
 All rights reserved.
 `

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- Implemented a new changelog.
+- Introduced a benchmarking script.
+
+### Fixed
+- Resolved issues with MPI installation.
+- Fixed the printing of configuration summaries with MPI.
+- Automated the process of adding a new kernel.
+- Improved packaging of software with CPack.
+- Addressed installation issues.
+- Fixed bivariate and trivariate kernel functionality.
```
Comment (SAbdulah) [inline CHANGELOG.md:17]: kernels

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- Corrected time-space kernel issues.
+
+### Changed
+- Updated the installation process for dependencies.
+- Modified the calculation of P for all kernels.
+- Adjusted CMake variables.
+- Revised the process of finding BlasPP and Catch2 libraries.
```
Comment (SAbdulah) [inline CHANGELOG.md:24]: BLASPP

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+- Updated doxygen documentation.
+
+### Removed
+- Eliminated non-stationary kernel support.
+- Removed Find OpenMP, LapackPP, and CuSolver.
```
Comment (SAbdulah) [inline CHANGELOG.md:29]: LAPACKPP --- CuSOLVER

File: cmake/ImportBlasPP.cmake [hunk 4]
```diff
@@ -4,51 +4,30 @@
 
 # @file ImportBlasPP.cmake
```
Comment (SAbdulah) [inline cmake/ImportBlasPP.cmake:5]: Could we rename this file to ImportBLASPP.cmake?

File: inst/include/kernels/Kernel.hpp [hunk 5]
```diff
@@ -140,15 +140,15 @@ namespace exageostat::kernels {
          * @return The value of P.
          *
          */
-        [[nodiscard]] int GetPValue() const;
+        [[nodiscard]] int GetP() const;
```
Comment (SAbdulah) [inline inst/include/kernels/Kernel.hpp:143]: P is the number of data variables. I think it is better to rename P to NumVars

File: src/data-units/DescriptorData.cpp [hunk 6]
```diff
@@ -70,7 +70,7 @@ void *DescriptorData<T>::GetRequest() {
     return this->mpRequest;
 }
 
-#ifdef EXAGEOSTAT_USE_HICMA
+#ifdef USE_HICMA
```
Comment (SAbdulah) [inline src/data-units/DescriptorData.cpp:73]: Will we use `USE_HICMA` instead of `EXAGEOSTAT_USE_HICMA` when building?

**Review by SAbdulah — CHANGES_REQUESTED** [review]

File: inst/include/kernels/Kernel.hpp (hunk 5 above)
Comment (mahmoudElkarargyBS) [reply inst/include/kernels/Kernel.hpp:143]: In the C there's P and number of parameters. Same In C++ so making P NumVars I think will be similar to num_params

else if (strcmp(arguments.kernel_fun, "univariate_spacetime_matern_stationary") == 0) {
        num_params = 7;
        p = 1;
    } else if (strcmp(arguments.kernel_fun, "bivariate_spacetime_matern_stationary") == 0) {
        num_params = 10;
        p = 2;

File: src/data-units/DescriptorData.cpp (hunk 6 above)
Comment (mahmoudElkarargyBS) [reply src/data-units/DescriptorData.cpp:73]: Indeed, we have eliminated all instances of "EXAGEOSTAT_" in the CMake variables. No need to prolong them unnecessarily.

File: inst/include/kernels/Kernel.hpp (hunk 5 above)
Comment (SAbdulah) [reply inst/include/kernels/Kernel.hpp:143]: params and vars are two different things. `Variables` here is the data variables, `parameters` is the model parameters. no we can use `num_params` and `num_vars`
